tests/Mutator/AbstractMutatorTestCase.php

```diff
@@ -32,22 +35,31 @@
      */
     protected $mutator;
 
-    public function doTest(string $inputCode, string $expectedCode = null): void
+    public function doTest(string $inputCode, $expectedCode = null): void
     {
+        $expectedCodeSamples = (array) $expectedCode;
+
         $inputCode = rtrim($inputCode, "\n");
 
         if ($inputCode === $expectedCode) {
             throw new \LogicException('Input code cant be the same as mutated code');
         }
 
-        $realMutatedCode = $this->mutate($inputCode);
+        $mutants = $this->mutate($inputCode);
 
-        if ($expectedCode !== null) {
-            $expectedCode = rtrim($expectedCode, "\n");
-            $this->assertSame($expectedCode, $realMutatedCode);
-            $this->assertSyntaxIsValid($realMutatedCode);
+        if ($expectedCode === null) {
+            $this->assertCount(0, $mutants);
         } else {
-            $this->assertSame($inputCode, $realMutatedCode);
+            foreach ($mutants as $realMutatedCode) {
+                $expectedCodeSample = array_shift($expectedCodeSamples);
+
+                if ($expectedCodeSample === null) {
```

Comment:
I think this will not error if there are more expected mutations than ones actual given. 
What about adding an assertSame on count of `$mutants` and `$expectedCodeSamples`

Reply:
fixed